org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -277,30 +277,30 @@ private void setFilePath(ScanBean bean) throws EventException {
 			bean.setFilePath(req.getFilePath());
 		}
 		logger.debug("Nexus file path set to {}", bean.getFilePath());
-		
+
 	}
-	
+
 	/**
 	 * Initialise the malcolm device with the point generator and the malcolm model
 	 * with its output directory. This needs to be done before validation as these values
-	 * are sent to the actual malcolm device over the connection for validation. 
+	 * are sent to the actual malcolm device over the connection for validation.
 	 * @param gen
 	 * @throws EventException
 	 * @throws ScanningException
 	 */
 	private void initializeMalcolmDevice(ScanBean bean, IPointGenerator<?> gen) throws EventException, ScanningException {
-		
+
 		ScanRequest<?> req = bean.getScanRequest();
-		
+
 		// check for a malcolm device, if one is found, set its output dir on the model
 		// and point generator on the malcolm device itself
 		if (bean.getFilePath() == null) return;
-		
+
 		String malcolmDeviceName = null;
 		MalcolmModel malcolmModel = null;
 		final Map<String, Object> detectorMap = req.getDetectors();
 		if (detectorMap == null) return;
-		
+
 		for (String detName : detectorMap.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)